### DISCUSSION TITLE
subtasks inherit categories from the parent

### DISCUSSION THREAD

**Discussion by bepolymathe** [TechbeeAT/jtxBoard#843] [Ideas]
Hi @patrickunterwegs 

The more I use jtx, the more I feel the need to have an option allowing subtasks to inherit categories from the parent. This would make project management easier. What do you think?
Have a nice day.

**Comment by patrickunterwegs** [comment]
Hey @bepolymathe !
Thanks, sounds like a useful enhancement. 

Let's break it down to some functions and behavioral points:

- I'd consider a dedicated setting to activate inheritance on all sub-entries (notes and tasks)
- the inheritance should be applied on all levels of sub-entries below
- existing entries are not migrated wegen the seeing is activated (otherwise it would get a bit complicated when you activate and deactivate the setting). Entries meat be updated manually in order to apply inheritance
- additional categories on sub-entries should not affect the parent

Would you agree with that? Do you find anything missing?

**Reply by bepolymathe** [reply]
Hi @patrickunterwegs 

I'm glad you think the idea makes sense. I totally agree with the points you make. 
> I'd consider a dedicated setting to activate inheritance on all sub-entries (notes and tasks)

I'm totally happy with this, although I imagine there will be a few users who want sub-tasks to inherit categories, but not sub-notes. 🤔 

> the inheritance should be applied on all levels of sub-entries below

Yes, I totally agree.

> existing entries are not migrated wegen the seeing is activated (otherwise it would get a bit complicated when you activate and deactivate the setting). Entries meat be updated manually in order to apply inheritance

Sounds reasonable.

> additional categories on sub-entries should not affect the parent

Yes, you're right, it's logical. What's more, if several sub-tasks have the same category but not their parent, a filter by category will reveal "horizontal" connections between certain projects. Interesting if you go back to the history of completed tasks, for example. 

Otherwise, I think you've covered everything. 
Thank you!

**Reply by patrickunterwegs** [reply]
@bepolymathe thank you very much for your donation and bist on Mastodon 🙂

And thanks for the discussion!
I'm reconsidering the above approach a bit. When you add subtasks from another app/platform, you might also end up having tasks that don't get the categories inherited.

I'm thinking if it's not better instead of physically adding the category to the sub-entries, to just mirror them in the UI and make sure they can be filtered and grouped (although this might get tricky). 

Or would you see a major advantage of the inherited categories are also physically saved for sub-entries?

I'll check if a split for a setting for subnotes and subtasks is easily possible.

**Reply by bepolymathe** [reply]
😉 

> Or would you see a major advantage of the inherited categories are also physically saved for sub-entries?

I use jtx on my mobile and thunderbird on my PC (linux). I'm therefore quite attached to finding categories on both sides because I mirror them "by hand" by assigning the same colors in jtx and thunderbird. So I find it limiting to have my categories only in jtx (as long as there's no desktop application). 

 > When you add subtasks from another app/platform, you might also end up having tasks that don't get the categories inherited.

That's true, but if I add several tasks in thunderbird with an A category, I'll be able to filter the tasks in jtx and link them to other "project" tasks afterwards. For me, this category will be "inbox" as a transitional category. We can certainly improve this...
